### PR TITLE
EfficientNetV2 Functional subclass

### PR DESCRIPTION
**Pull request by IMvision12**
https://colab.research.google.com/drive/1YDJTN4P4LArfWjXGW1u7ke_TTL5a8SfB?usp=sharing
Fix : #1425
@ianstenbit

### CODE REVIEW

**Review by ianstenbit — APPROVED** [review]
Looks good -- thank you!

**Comment by ianstenbit** [comment]
Test failure looks real. Please take a look -- thanks!
```
blocks = float(sum(args["num_repeat"] for args in blocks_args))
KeyError: 'num_repeat'
```

**Comment by IMvision12** [comment]
```
blocks_args = DEFAULT_BLOCKS_ARGS["efficientnetv2-b0"]
blocks = float(sum(args["num_repeat"] for args in blocks_args))
print(blocks)
```

Not sure why the test is failing, I tested the above code and it works fine

**Comment by ianstenbit** [comment]
@IMvision12 based on the error message (are you able to see it?), this is related to serialization / `from_config`. I'd guess the `blocks_args` param isn't being serialized correctly (possibly because in this case what's passed in is "default" but what we serialize is the full dictionary -- we could fix this by making a copy of the input `blocks_args` param and store that on `self.blocks_args`)

You should be able to run this test locally with PyTest to test further.

**Comment by IMvision12** [comment]
Oh, yeah i didn't see that

**Comment by IMvision12** [comment]
```
=========================================== test session starts ===========================================
platform win32 -- Python 3.9.10, pytest-7.2.2, pluggy-1.0.0
rootdir: F:\effi\keras-cv, configfile: setup.cfg
collected 6 items

keras_cv\models\efficientnet_v2_test.py .....s                                                       [100%]

====================================== 5 passed, 1 skipped in 14.60s ======================================
```

File: keras_cv/models/efficientnet_v2.py [hunk 1]
```diff
@@ -637,189 +610,258 @@ def EfficientNetV2(
...
+    ):
+        if blocks_args == "default":
+            blocks_args = DEFAULT_BLOCKS_ARGS[model_name]
 
-    if weights and not tf.io.gfile.exists(weights):
-        raise ValueError(
-            "The `weights` argument should be either `None` or the path to the "
-            "weights file to be loaded. Weights file not found at location: {weights}"
-        )
+        copy_blocks_args = copy.deepcopy(blocks_args)
```
Comment (ianstenbit) [inline keras_cv/models/efficientnet_v2.py:654]: nit: maybe s/copy_blocks_args/input_blocks_args?

Comment (IMvision12) [reply keras_cv/models/efficientnet_v2.py:654]: name change?

Comment (IMvision12) [reply keras_cv/models/efficientnet_v2.py]: Done.

**Review by ianstenbit — APPROVED** [review]

**Comment by IMvision12** [comment]
@ianstenbit, would we be interested in integrating SSD (Single Shot Detector) models to kcv? I was thinking about working on it.

**Comment by ianstenbit** [comment]
> @ianstenbit, would we be interested in integrating SSD (Single Shot Detector) models to kcv? I was thinking about working on it.

I'd open an issue about it and tag LukeWood -- I think in general new OD models are something we're interested in, but prioritizing YOLOx and YOLOv8 first.

**Pull request merged**